Loop/Managers/Alerts/StoredAlert.swift

```diff
@@ -145,6 +149,48 @@ extension Alert.Trigger {
     }
 }
 
+extension Alert.InterruptionLevel {
+    enum StorageError: Error {
+        case invalidStoredLevel
+    }
+    
+    var storedValue: Int16 {
+        // Since this is arbitrary anyway, might as well make it match iOS's values
+        switch self {
+        case .active:
+            if #available(iOS 15.0, *) {
+                return Int16(UNNotificationInterruptionLevel.active.rawValue)
+            } else {
+                return 1
+            }
+        case .timeSensitive:
+            if #available(iOS 15.0, *) {
+                return Int16(UNNotificationInterruptionLevel.timeSensitive.rawValue)
+            } else {
+                return 2
+            }
+        case .critical:
+            if #available(iOS 15.0, *) {
+                return Int16(UNNotificationInterruptionLevel.critical.rawValue)
+            } else {
+                return 3
+            }
+        }
+    }
+    
+    init(storedValue: Int16) throws {
```

Comment:
A throw seems pretty heavy-handed here. The typical usage in Loop is to make it a failable initializer (`init?`).

Reply:
heh, I went back and forth on this...I thought you'd like this better.  I guessed wrong :).

Reply:
What I don't like is silently failing when failable initializers return `nil` during deserialization. If the key is found in the dictionary, but then deserialization of the value fails, then that is an error. Unfortunately, most of our deserialization code (typically via RawRepresentable), silently fails. 

Otherwise I'm fine with failable initializers and personally tend to steer away from throwing in the Loop code.

That's just my two cents, though...

Reply:
My preference when decoding is to throw, because sometimes the type is used as an optional field.  Now we've got a situation: is the initialization failing due a coding error or is the field value actually nil?  So now all the callers (which deal with that field) need to check for nil, which means "coding error", which is not the same as the (optional) field just not being present.  

In other words, you can no longer do:

```
   field = optionalRawValue.map { MyEnum(from: $0) }
```

Because the `MyEnum` constructor might return nil.

Again, I also think this is all philosophical but in a system when I can influence the style guides, that's my preferred way to go.  Just my 2c.

Reply:
I understand. Just note that this style isn't used in any of the rest of the Loop code (excepting the Decoder protocol conformance).